addons/product/models/product_template.py

```diff
@@ -175,6 +175,7 @@ def _compute_currency_id(self):
     def _compute_cost_currency_id(self):
         self.cost_currency_id = self.env.company.currency_id.id
 
+    @api.depends_context('pricelist', 'partner', 'quantity', 'uom', 'date', 'no_variant_attributes_price_extra')
```

Comment:
I am not a big fan of this change.
Especially since in 15.3 we cleaned up code related to context over here: https://github.com/odoo/odoo/pull/82374
Which means we will have to find a solution for that during the forward-ports either way.

It's a big stretch since this was such a long time ago but do you know why it's necessary?